### ISSUE TITLE
isinstance & Callable

### ISSUE THREAD

**Issue by cgarciae** [microsoft/pylance-release#247]
## Environment data
-   Language Server version: Pylance language server 2020.8.1 (pyright 0916fb9e)
-   OS and version: Ubuntu 20.04
-   Python version: 3.8.1

## Actual behaviour

Recently I am getting a type issue when using `isinstance` to check for callables:

```python
import typing as tp

def f():
    ...

print(isinstance(f, tp.Callable)) # Second argument to "isinstance" must be a class or tuple of classes
# TypeVar or generic type with type arguments is not allowedPylance (reportGeneralTypeIssues)
```

According to PEP 484 this should be supported:

> Because typing.Callable does double-duty as a replacement for collections.abc.Callable , isinstance(x, typing.Callable) is implemented by deferring to `isinstance(x, collections.abc.Callable) . However, isinstance(x, typing.Callable[...]) is not supported.

Previous versions of Pylance where ok with this type of code.

**Comment by erictraut** [comment]
Thanks for the bug report. This was a regression related to a recent change I made. I've fixed the problem and augmented the unit tests to prevent similar regressions in the future. The fix will be in the next version of pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.8.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202082-20-august-2020
